### PR TITLE
Entity reference widget doesn't take field cardinality limitations into account

### PR DESCRIPTION
**Pull request by VD39**
## Issue

https://github.com/jsdrupal/drupal-admin-ui/issues/282

## Screenshot / UI changes

Added a check to see if the current widget supports single value.

I couldn't find the max value in the schema, so this PR assumes there are an unlimited number of selection that can be made.

## Testing instructions

Test author (single value) and tags (multiple value).

### CODE REVIEW

**Comment by dawehner** [comment]
Seems like a perfect fit. This ensures both that the suggestions aren't requested and potential suggestions returned by some outstanding request, aren't leaked in

**Comment by dawehner** [comment]
> I couldn't find the max value in the schema, so this PR assumes there are an unlimited number of selection that can be made.

Well, it is the same as for the file reference, so maybe we should deal with both cases at the same time

File: src/components/02_atoms/Widgets/EntityReferenceAutocomplete.js [hunk 1]
```diff
@@ -147,13 +151,30 @@ class EntityReferenceAutocomplete extends React.Component {
       );
   };
 
+  cannotAddMoreEntities = () => {
```
Comment (lauriii) [inline src/components/02_atoms/Widgets/EntityReferenceAutocomplete.js]: This is very nitpicky but the name of this function sounds a bit strange to me. How about just changing this to `getMaxItems` and then doing the comparison in the calling code?

Comment (VD39) [reply src/components/02_atoms/Widgets/EntityReferenceAutocomplete.js]: Sure, although the function does two checks, if is single and has an item, or if the max item has been entered. How about changing it to maxItemsSelected?

Comment (dawehner) [reply src/components/02_atoms/Widgets/EntityReferenceAutocomplete.js]: I like @lauriii's idea, given we might want to render some description like: "N items left"

Comment (VD39) [reply src/components/02_atoms/Widgets/EntityReferenceAutocomplete.js]: I have updated the method, I used the same method from the file upload widget to check the maxItems.

**Pull request merged**